docs/achitecture_and_design/api_design.md

```diff
@@ -0,0 +1,121 @@
+# API DESIGN
+
+The following are the RESTful endpoints needed to support basic chat and follow-up creation, retrieval, updating, and deletion.
+
+## Chats
+
+### Retrieve individual chat
+`GET /api/chats/<id>`
+```json
+{
+    "id": 2,
+    "user": {
+        "id": 1,
+        "name": "..."
+    },
+    "title": "...",
+    "description": "...",
+    "date": "...",
+    "follow_ups_url": "api/chats/<id>/followups",
+    "follow_up_count": 5
+}
+```
+
+### Retrieve a user's chats
+`GET /api/user/2/chats`
+```json
+{
+    "count": 123,
+    "next": "...url to next page of results...",
+    "previous": "...url to next previous of results...",
+    "chats": [
+        {
+            "id": 1,
+            "user": {
+                "id": 1,
+                "name": "..."
+            },
+            "title": "...",
+            "description": "...",
+            "date": "...",
+            "follow_ups_url": "api/chats/<id>/followups",
+            "follow_up_count": 5
+        },
+        ...
+    ]
+}
+```
+
+### Retrive all chats
+`GET /api/chats`
+(Same data returned as User's Chats)
+
+
+### Create a new chat
+`POST /api/chats`
+```json
+{
+    "title": "...",
+    "date": "...",
+    "description": "..."
+}
+```
+
+### Update a chat
+`PUT /api/chats/<id>`
+(Same payload as POST)
+
+### Delete a chat
+`DELETE /api/chats/<id>`
+
+
+## Follow-ups
+
+### Retrieve follow-ups for a particular chat
+`GET /api/chats/<id>/followups`
+```json
+{
+    "count": 123,
+    "next": "...url to next page of results...",
+    "previous": "...url to next previous of results...",
+    "follow_ups": [
+        {
+            "user": {
+                "id": 1,
+                "name": "..."
+            },
+            "date": "...",
+            "content": "...",
+            "penny_chat": "api/chats/<id>"
```

Comment:
Not just the `<id>`?

Reply:
It's not obvious with the way we've written it, but all the relationships will be represented with hyperlinks (like in the github v3; ex: api https://developer.github.com/v3/issues/#response ).